src/model/modifier/DraftModifier.js

```diff
@@ -204,13 +204,41 @@ var DraftModifier = {
     selectionState: SelectionState,
     blockType: DraftBlockType
   ): ContentState {
-    return setBlockTypeForContentState(
+    const op = block => block.merge({type: blockType, depth: 0});
```

Comment:
Nice! This is an elegant way to reuse the transaction function.

Nits:

- Internally we've been trying to use parens even on single-argument fat-arrow functions. The quick return syntax is fine, though.
- I'd like `operation` better than `op`. :)